stress/shell/src/main/java/alluxio/stress/cli/fuse/FuseIOBench.java

```diff
@@ -0,0 +1,365 @@
+        mFilesPath.add(mParameters.mLocalPath + "/data-" + i);
+      }
+
+      mBuffer = new byte[(int) FormatUtils.parseSpaceSize(mParameters.mBufferSize)];
+      Arrays.fill(mBuffer, (byte) 'A');
+
+      mFileSize = FormatUtils.parseSpaceSize(mParameters.mFileSize);
+    }
+
+    @Override
+    public Void call() {
+      try {
+        runInternal();
+      } catch (Exception e) {
+        LOG.error(Thread.currentThread().getName() + ": failed", e);
+        mThreadCountResult.addErrorMessage(e.getMessage());
+      }
+
+      // Update thread count result
+      mThreadCountResult.setEndMs(CommonUtils.getCurrentMs());
+      mContext.mergeThreadResult(mThreadCountResult);
+
+      return null;
+    }
+
+    private void runInternal() throws Exception {
+      // When to start recording measurements
+      long recordMs = mContext.getStartMs() + FormatUtils.parseTimeSize(mParameters.mWarmup);
+      mThreadCountResult.setRecordStartMs(recordMs);
+      boolean isRead = FuseIOOperation.isRead(mParameters.mOperation);
+
+      long waitMs = mContext.getStartMs() - CommonUtils.getCurrentMs();
+      if (waitMs < 0) {
+        throw new IllegalStateException(String.format(
+            "Thread missed barrier. Set the start time to a later time. start: %d current: %d",
+            mContext.getStartMs(), CommonUtils.getCurrentMs()));
+      }
+      CommonUtils.sleepMs(waitMs);
+      mStartBarrierPassed = true;
+
+      for (int i = 0; i < mFilesPath.size(); i++) {
+        String filePath = mFilesPath.get(i);
+        while (!Thread.currentThread().isInterrupted() && (!isRead
+            || CommonUtils.getCurrentMs() < mContext.getEndMs())) {
+          long ioBytes = applyOperation(filePath);
+
+          long currentMs = CommonUtils.getCurrentMs();
+          // Start recording after the warmup
+          if (currentMs > recordMs) {
+            if (ioBytes > 0) {
+              mThreadCountResult.incrementIOBytes(ioBytes);
+              // Writing one file is done in one applyOperation call; start writing next file
+              if (!isRead) {
+                break;
+              }
+            } else if (i == mFilesPath.size() - 1) {
+              // Done reading the last file. Finish too early.
+              throw new IllegalArgumentException(String.format("Thread %d finishes reading all "
+                  + "its files before the bench ends. For more accurate result, use more files, "
+                  + "or larger files, or shorter durations."));
+            } else {
+              // Done reading this file.
+              break;
+            }
+          }
+        }
+      }
+    }
+
+    private long applyOperation(String filePath) throws IOException {
+      if (FuseIOOperation.isRead(mParameters.mOperation) && mInStream == null) {
+        mInStream = new FileInputStream(filePath);
+      }
+      switch (mParameters.mOperation) {
+        case READ: {
```

Comment:
Keep read and write operation the same pattern in the beginning.
If write writes the whole file, then read reads the whole file.
If write writes up to buffer size, then read reads up to buffer size.

Reply:
or we can have readArray and readFully operations

Reply:
They now read/write up to the buffer size, no whole file anymore (for now). The reading loop is by calling `applyOpeartion()` and the writing loop is inside `applyOpeartion()`, which makes it a little confusing. I have to keep a current offset for writing but not for reading and that's why they are separated. I may have to move the time check (getCurrentMS vs getEndMs) inside `applyOpeartion()` to make the logic more straightforward.

Reply:
Oh, if that's the case we can do write and read logic separately to make the for & while loop logics easier